public_html/wp-content/plugins/wordcamp-speaker-feedback/includes/form.php

```diff
@@ -17,16 +27,52 @@ function has_feedback_form() {
 
 /**
  * Short-circuit the content, and output the feedback form (or the session select).
+ *
+ * This assumes that `wcb_session` is the only post type supporting speaker feedback. This will need to be updated
+ * if more post type support is added.
+ *
+ * @global WP_Post $post
+ *
+ * @return string
  */
 function render( $content ) {
+	global $post;
+
+	$now = date_create( time(), wp_timezone() );
+
 	if ( has_feedback_form() ) {
+		$accepts_feedback = post_accepts_feedback( $post->ID );
+
+		if ( is_wp_error( $accepts_feedback ) ) {
+			$message = $accepts_feedback->get_error_message();
+			$file    = 'form-not-available.php';
+		} else {
+			$file = 'form-feedback.php';
+		}
+
 		ob_start();
-		require get_views_path() . 'form-feedback.php';
-		return $content . ob_get_clean();
+		require get_views_path() . $file;
+
+		$content = $content . ob_get_clean(); // Append form to the normal content.
 	} elseif ( is_page( get_option( OPTION_KEY ) ) ) {
+		$wordcamp   = get_wordcamp_post();
+		$start_date = $wordcamp->meta['Start Date (YYYY-mm-dd)'][0] ?? $now->getTimestamp() + YEAR_IN_SECONDS;
```

Comment:
I think the fact that the start date is not the real UTC value will bite some of the UTC+ timezones — they might have morning sessions, but `/feedback` will say the event hasn't started. We could convert this value into the site's timezone, since that has a good chance of being correct. Something like

```php
$date_string = date( 'Y-m-d', $wordcamp->meta['Start Date (YYYY-mm-dd)'][0] );
$start_date  = date_create( $date_string, wp_timezone() )->getTimestamp();
```

Reply:
> Why default this to a year from now, so it's an arbitrary future date, or is the year important?

The thinking was that if the site doesn't have a meta value set for the start date, it probably hasn't been scheduled yet, in which case, the correct message would be `Feedback forms are not available until the event has started`. If the value defaulted to `0`, the message shown would be `Feedback forms are closed for this event`, which doesn't seem as accurate. The fact that it's a year ahead is arbitrary, it just needs to be far enough ahead that it won't fall into any window of time that we may set in the future.

Reply:
> I think the fact that the start date is not the real UTC value will bite some of the UTC+ timezones

Good call. I expanded on your suggested here: 223b0f61

Reply:
Looks good, and you were able to add comments which addressed my initial  🤔❓